src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -20,29 +21,45 @@ public class DeleteCommand extends Command {
 
     public static final String MESSAGE_USAGE = COMMAND_WORD
             + ": Deletes the person identified by the index number used in the displayed person list.\n"
-            + "Parameters: INDEX (must be a positive integer)\n"
-            + "Example: " + COMMAND_WORD + " 1";
+            + "Parameters: INDICES (must be a positive integer or a range, separated by commas)\n"
+            + "Example: " + COMMAND_WORD + " 1 2 3 5-7";
 
-    public static final String MESSAGE_DELETE_PERSON_SUCCESS = "Deleted Person: %1$s";
+    public static final String MESSAGE_DELETE_PERSON_SUCCESS = "Deleted Entries(s): \n";
 
-    private final Index targetIndex;
+    private final List<Index> targetIndices;
 
-    public DeleteCommand(Index targetIndex) {
-        this.targetIndex = targetIndex;
+    public DeleteCommand(List<Index> targetIndices) {
+        this.targetIndices = targetIndices;
     }
 
+    private void validateIndices(List<Index> targetIndices, Integer lastShownListSize) throws CommandException {
+        for (Index targetIndex : targetIndices) {
+            if (targetIndex.getZeroBased() >= lastShownListSize) {
+                throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+            }
+        }
+    }
+
+    private void deletePersonsFromModel(Model model, List<Person> personToDeleteList) {
+        for (Person personToDelete : personToDeleteList) {
+            model.deletePerson(personToDelete);
+        }
+    }
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         List<Person> lastShownList = model.getPersonList();
-
-        if (targetIndex.getZeroBased() >= lastShownList.size()) {
-            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        List<Person> personToDeleteList = new ArrayList<>();
+        String commandResultString = MESSAGE_DELETE_PERSON_SUCCESS;
+        validateIndices(targetIndices, lastShownList.size());
+        for (Index targetIndex : targetIndices) {
+
+            Person personToDelete = lastShownList.get(targetIndex.getZeroBased());
+            personToDeleteList.add(personToDelete);
+            commandResultString += Messages.format(personToDelete) + "\n";
```

Comment:
Might want to use `StringBuilder` for more efficient concat? Could potentially take a long time if user is deleting a lot of contacts at once.